env_spec.py

```diff
@@ -19,138 +28,204 @@
 def create_list(input_str):
     """
     Creates a list from input_str_list.Returns the new list.
+    If input_str=[1,2,], returns empty list(syntax error).
     """
     input_str = input_str.replace("[", "")
     input_str = input_str.replace("]", "")
     new_list = input_str.split(",")
 
+    if "" in new_list:
+        return []
+
     for pos in range(0, len(new_list)):
-        new_list[pos] = str(new_list[pos])
+        new_list[pos] = str(new_list[pos]).strip()
 
     return new_list
 
 
-def check_input(input_field):
+def render_label(env_spec_entry):
     """
-    Checks syntax of input field(uppercase latin chars,numeric digits, underscores).
-    Returns False for error or True for correct syntax.
+    Creates the html output for label.Takes as argument the current dictionary.
     """
-    alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
+    label_str = f'<label for="env_spec_{env_spec_entry["name"].lower()}">{env_spec_entry["name"]}</label>\n'
 
-    ret = re.match(alphanumeric_that_does_not_start_with_digit, input_field)
+    return label_str
 
-    if ret is None:
-        return False
 
-    return True
+def render_input(env_spec_entry):
+    """
+    Creates the html output for env_spec_entry input, which is the current dictionary.
+    """
+    if env_spec_entry["default_value"] is None:
+        input_str = f'<input id="env_spec_{env_spec_entry["name"].lower()}" name="{env_spec_entry["name"].lower()}" type="{env_spec_entry["type"]}" />\n'
+    else:
+        input_str = f'<input id="env_spec_{env_spec_entry["name"].lower()}" name="{env_spec_entry["name"].lower()}" type="{env_spec_entry["type"]}" value="{env_spec_entry["default_value"]}" />\n'
```

Comment:
Since most of the text between the input fields with or without default value is the same, I propose changing this to something more Pythonic to [avoid repeating ourselves](
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself):

Something like the following works:

```py
default_value = env_spec_entry["default_value"]
default_value_state = f'value="{default_value}"' if default_value else ""
env_spec_entry_input = (
    f'<input id="env_spec_{env_spec_entry["name"].lower()}" '
    f'name="{env_spec_entry["name"].lower()}" type="{env_spec_entry["type"]}" '
    f'{default_value_state}" />\n'
)
return env_spec_entry_input
```